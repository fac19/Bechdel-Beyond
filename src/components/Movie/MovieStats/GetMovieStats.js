import React from 'react';
import Typography from '@material-ui/core/Typography';

export function GetMovieStatsCast({ statsCastData }) {
	if (!statsCastData) return <h1>Locading</h1>;
	return (
		<>
			<Typography variant="h5" component="h3">
				Cast gender balance:
			</Typography>
			<Typography variant="h5" component="h3">
				Female: {statsCastData.gender_parity_cast.female}
			</Typography>
			<Typography variant="h5" component="h3">
				Male: {statsCastData.gender_parity_cast.male}
			</Typography>
			<Typography variant="h5" component="h3">
				Not declared: {statsCastData.gender_parity_cast.notlisted}
			</Typography>
		</>
	);
}

export function GetMovieStatsCrew({ statsCrewData }) {
	if (!statsCrewData) return <h1>Loading</h1>;

	return (
		<>
			<Typography variant="h5" component="h3">
				Crew gender balance:
			</Typography>
			<Typography variant="h5" component="h3">
				Female: {statsCrewData.gender_parity.female}
			</Typography>
			<Typography variant="h5" component="h3">
				Male: {statsCrewData.gender_parity.male}
			</Typography>
			<Typography variant="h5" component="h3">
				Not declared: {statsCrewData.gender_parity.notlisted}
			</Typography>
		</>
	);
}
