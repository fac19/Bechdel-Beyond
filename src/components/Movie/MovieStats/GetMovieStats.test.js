import React from 'react';
import renderer from 'react-test-renderer';

import { GetMovieStatsCast, GetMovieStatsCrew } from './GetMovieStats';

it('should render <lodading> in csst stats section if no data', () => {
	const stats = renderer.create(<GetMovieStatsCast />).toJSON();
	expect(stats).toMatchSnapshot();
});

it('should render cast stats section correctly if data provided', () => {
	const statsCastData = {
		id: 16,
		gender_parity_cast: {
			male: 51,
			female: 27,
			notlisted: 38,
		},
		movapi_id: 597,
	};

	const stats = renderer
		.create(<GetMovieStatsCast statsCastData={statsCastData} />)
		.toJSON();
	expect(stats).toMatchSnapshot();
});

it('should render <lodading> in crew stats section if no data', () => {
	const stats = renderer.create(<GetMovieStatsCrew />).toJSON();
	expect(stats).toMatchSnapshot();
});

it('should render crew stats section correctly if data provided', () => {
	const statsCrewData = {
		id: 1,
		director: 'James',
		assistant_director: 'Ako',
		producer: 'Gio',
		gender_parity: {
			male: 2,
			female: 3,
			notlisted: 34,
		},
		movapi_id: 34,
	};

	const stats = renderer
		.create(<GetMovieStatsCrew statsCrewData={statsCrewData} />)
		.toJSON();
	expect(stats).toMatchSnapshot();
});
