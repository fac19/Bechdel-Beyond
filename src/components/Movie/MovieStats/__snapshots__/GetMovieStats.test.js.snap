// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cast stats section renders correctly on movie page 1`] = `
<h1>
  Locading
</h1>
`;

exports[`Cre stats section renders correctly on movie page 1`] = `
<h1>
  Loading
</h1>
`;
